The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17928268371/job/50979752318) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/50979752318) [(plain)](https://github.com/rust-lang/rust/commit/1932ce333ad0bb83b84b695db3ee03fc012ca30e/checks/50979752318/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

---- [codegen] tests/codegen-llvm/global-allocator-attributes.rs stdout ----
------FileCheck stdout------------------------------

------FileCheck stderr------------------------------
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:13:17: error: CHECK-NEXT: expected string not found in input
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc(i[[SIZE:[0-9]+]] {{.*}}%size, i[[SIZE]] allocalign{{.*}} %align)
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:50:95: note: scanning from here
; Function Attrs: cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0)
                                                                                              ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:51:57: note: possible intended match here
define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc12___rust_alloc(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #1 {
                                                        ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:20:79: error: undefined variable: SIZE
 // CHECK-NEXT: define void @{{.*}}__rust_dealloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] {{.*}} %align)
                                                                              ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:20:103: error: undefined variable: SIZE
 // CHECK-NEXT: define void @{{.*}}__rust_dealloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] {{.*}} %align)
                                                                                                      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:63:36: note: possible intended match here
define dso_local void @_RNvCs59eiacZUMsN_7___rustc14___rust_dealloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 noundef %align) unnamed_addr #2 {
                                   ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:92: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                           ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:116: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                                                   ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:151: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                                                                                      ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:34:72: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc_zeroed(i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align)
                                                                       ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:34:96: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc_zeroed(i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align)
                                                                                               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:87:54: note: possible intended match here
define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc19___rust_alloc_zeroed(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #4 {
                                                     ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll
Check file: /checkout/tests/codegen-llvm/global-allocator-attributes.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
           1: ; ModuleID = 'global_allocator_attributes.9107941623879d1c-cgu.0' 
           2: source_filename = "global_allocator_attributes.9107941623879d1c-cgu.0" 
           3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-i128:128-n32:64-S128-ni:1:10:20" 
           4: target triple = "wasm32-unknown-wasip1" 
           5:  
           6: @alloc_476509aa7047bd805d630c6546685797 = private unnamed_addr constant [14 x i8] c"explicit panic", align 1 
           7: @alloc_4adda3ed1088f74f07ee43568cdd9f39 = private unnamed_addr constant [60 x i8] c"/checkout/tests/codegen-llvm/global-allocator-attributes.rs\00", align 1 
           8: @alloc_b770eaa3f4771da96efd94be0aca7ca5 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\0E\00\00\00\0D\00\00\00" }>, align 4 
           9: @alloc_a5aa75597376cd18f9df6ea9fddc2db2 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\15\00\00\00\0D\00\00\00" }>, align 4 
          10: @alloc_fea0b74bc1f394cd9f32c70385fabd3f = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\1C\00\00\00\0D\00\00\00" }>, align 4 
          11: @alloc_672a73527c968cb7e9717bc22e580e47 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00#\00\00\00\0D\00\00\00" }>, align 4 
          12:  
          13: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::alloc 
          14: ; Function Attrs: cold noreturn nounwind 
          15: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$5alloc17h1fa54a43d9d55781E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          16: start: 
          17: ; call std::panicking::begin_panic 
          18:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_b770eaa3f4771da96efd94be0aca7ca5) #7 
          19:  unreachable 
          20: } 
          21:  
          22: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::dealloc 
          23: ; Function Attrs: cold noreturn nounwind 
          24: define dso_local void @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$7dealloc17hb5c6154adcc1b844E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, ptr noundef readnone captures(none) %ptr, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          25: start: 
          26: ; call std::panicking::begin_panic 
          27:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_a5aa75597376cd18f9df6ea9fddc2db2) #7 
          28:  unreachable 
          29: } 
          30:  
          31: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::realloc 
          32: ; Function Attrs: cold noreturn nounwind 
          33: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$7realloc17hb1272095cf277bc4E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, ptr noundef readnone captures(none) %ptr, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1, i32 noundef %new_size) unnamed_addr #0 { 
          34: start: 
          35: ; call std::panicking::begin_panic 
          36:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_fea0b74bc1f394cd9f32c70385fabd3f) #7 
          37:  unreachable 
          38: } 
          39:  
          40: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::alloc_zeroed 
          41: ; Function Attrs: cold noreturn nounwind 
          42: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$12alloc_zeroed17h48656aee732bdc33E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          43: start: 
          44: ; call std::panicking::begin_panic 
          45:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_672a73527c968cb7e9717bc22e580e47) #7 
          46:  unreachable 
          47: } 
          48:  
          49: ; __rustc::__rust_alloc 
          50: ; Function Attrs: cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0) 
next:13'0                                                                                                   X error: no match found
          51: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc12___rust_alloc(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #1 { 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:13'1                                                             ?                                                                                                     possible intended match
          52: start: 
next:13'0     ~~~~~~~
          53:  %0 = add i32 %align, -1 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          54:  %1 = icmp sgt i32 %0, -1 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          55:  tail call void @llvm.assume(i1 %1) 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          56: ; call std::panicking::begin_panic 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          57:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_b770eaa3f4771da96efd94be0aca7ca5) #7 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          58:  unreachable 
next:13'0     ~~~~~~~~~~~~~
          59: } 
next:13'0     ~~
          60:  
next:13'0     ~
          61: ; __rustc::__rust_dealloc 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          62: ; Function Attrs: cold noreturn nounwind allockind("free") 
next:20'0                                                               X error: match failed for invalid pattern
next:20'1                                                                 undefined variable: SIZE
next:20'2                                                                 undefined variable: SIZE
          63: define dso_local void @_RNvCs59eiacZUMsN_7___rustc14___rust_dealloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 noundef %align) unnamed_addr #2 { 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:20'3                                        ?                                                                                                                                             possible intended match
          64: start: 
next:20'0     ~~~~~~~
          65:  %0 = add i32 %align, -1 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          66:  %1 = icmp sgt i32 %0, -1 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          67:  tail call void @llvm.assume(i1 %1) 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          68: ; call std::panicking::begin_panic 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          69:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_a5aa75597376cd18f9df6ea9fddc2db2) #7 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          70:  unreachable 
next:20'0     ~~~~~~~~~~~~~
          71: } 
next:20'0     ~~
          72:  
next:20'0     ~
          73: ; __rustc::__rust_realloc 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          74: ; Function Attrs: cold noreturn nounwind allockind("realloc,aligned") allocsize(3) 
next:27'0                                                                                       X error: match failed for invalid pattern
next:27'1                                                                                         undefined variable: SIZE
next:27'2                                                                                         undefined variable: SIZE
next:27'3                                                                                         undefined variable: SIZE
          75: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc14___rust_realloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 allocalign noundef %align, i32 noundef %new_size) unnamed_addr #3 { 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          76: start: 
next:27'0     ~~~~~~~
          77:  %0 = add i32 %align, -1 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          78:  %1 = icmp sgt i32 %0, -1 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          79:  tail call void @llvm.assume(i1 %1) 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          80: ; call std::panicking::begin_panic 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          81:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_fea0b74bc1f394cd9f32c70385fabd3f) #7 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          82:  unreachable 
next:27'0     ~~~~~~~~~~~~~
          83: } 
next:27'0     ~~
          84:  
next:27'0     ~
          85: ; __rustc::__rust_alloc_zeroed 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          86: ; Function Attrs: cold noreturn nounwind allockind("alloc,zeroed,aligned") allocsize(0) 
next:34'0                                                                                            X error: match failed for invalid pattern
next:34'1                                                                                              undefined variable: SIZE
next:34'2                                                                                              undefined variable: SIZE
          87: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc19___rust_alloc_zeroed(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #4 { 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:34'3                                                          ?                                                                                                               possible intended match
          88: start: 
next:34'0     ~~~~~~~
          89:  %0 = add i32 %align, -1 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          90:  %1 = icmp sgt i32 %0, -1 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          91:  tail call void @llvm.assume(i1 %1) 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          92: ; call std::panicking::begin_panic 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          93:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_672a73527c968cb7e9717bc22e580e47) #7 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          94:  unreachable 
next:34'0     ~~~~~~~~~~~~~
          95: } 
next:34'0     ~~
          96:  
next:34'0     ~
          97: ; std::panicking::begin_panic 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          98: ; Function Attrs: cold minsize noinline noreturn nounwind optsize 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          99: declare dso_local void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance), i32 noundef, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16)) unnamed_addr #5 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         100:  
next:34'0     ~
         101: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         102: declare void @llvm.assume(i1 noundef) #6 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         103:  
next:34'0     ~
         104: attributes #0 = { cold noreturn nounwind "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         105: attributes #1 = { cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         106: attributes #2 = { cold noreturn nounwind allockind("free") "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         107: attributes #3 = { cold noreturn nounwind allockind("realloc,aligned") allocsize(3) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         108: attributes #4 = { cold noreturn nounwind allockind("alloc,zeroed,aligned") allocsize(0) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         109: attributes #5 = { cold minsize noinline noreturn nounwind optsize "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         110: attributes #6 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         111: attributes #7 = { noreturn nounwind } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         112:  
next:34'0     ~
         113: !llvm.ident = !{!0} 
next:34'0     ~~~~~~~~~~~~~~~~~~~~
         114:  
next:34'0     ~
         115: !0 = !{!"rustc version 1.92.0-nightly (1932ce333 2025-09-22)"} 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>

------------------------------------------

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll" "/checkout/tests/codegen-llvm/global-allocator-attributes.rs" "--check-prefix=CHECK" "--allow-unused-prefixes" "--dump-input-context" "100"
stdout: none
--- stderr -------------------------------
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:13:17: error: CHECK-NEXT: expected string not found in input
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc(i[[SIZE:[0-9]+]] {{.*}}%size, i[[SIZE]] allocalign{{.*}} %align)
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:50:95: note: scanning from here
; Function Attrs: cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0)
                                                                                              ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:51:57: note: possible intended match here
define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc12___rust_alloc(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #1 {
                                                        ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:20:79: error: undefined variable: SIZE
 // CHECK-NEXT: define void @{{.*}}__rust_dealloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] {{.*}} %align)
                                                                              ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:20:103: error: undefined variable: SIZE
 // CHECK-NEXT: define void @{{.*}}__rust_dealloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] {{.*}} %align)
                                                                                                      ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:63:36: note: possible intended match here
define dso_local void @_RNvCs59eiacZUMsN_7___rustc14___rust_dealloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 noundef %align) unnamed_addr #2 {
                                   ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:92: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                           ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:116: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                                                   ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:27:151: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
                                                                                                                                                      ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:34:72: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc_zeroed(i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align)
                                                                       ^
/checkout/tests/codegen-llvm/global-allocator-attributes.rs:34:96: error: undefined variable: SIZE
 // CHECK-NEXT: define noalias{{.*}} ptr @{{.*}}__rust_alloc_zeroed(i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align)
                                                                                               ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll:87:54: note: possible intended match here
define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc19___rust_alloc_zeroed(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #4 {
                                                     ^

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/codegen-llvm/global-allocator-attributes/global-allocator-attributes.ll
Check file: /checkout/tests/codegen-llvm/global-allocator-attributes.rs

-dump-input=help explains the following input dump.

Input was:
<<<<<<
           1: ; ModuleID = 'global_allocator_attributes.9107941623879d1c-cgu.0' 
           2: source_filename = "global_allocator_attributes.9107941623879d1c-cgu.0" 
           3: target datalayout = "e-m:e-p:32:32-p10:8:8-p20:8:8-i64:64-i128:128-n32:64-S128-ni:1:10:20" 
           4: target triple = "wasm32-unknown-wasip1" 
           5:  
           6: @alloc_476509aa7047bd805d630c6546685797 = private unnamed_addr constant [14 x i8] c"explicit panic", align 1 
           7: @alloc_4adda3ed1088f74f07ee43568cdd9f39 = private unnamed_addr constant [60 x i8] c"/checkout/tests/codegen-llvm/global-allocator-attributes.rs\00", align 1 
           8: @alloc_b770eaa3f4771da96efd94be0aca7ca5 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\0E\00\00\00\0D\00\00\00" }>, align 4 
           9: @alloc_a5aa75597376cd18f9df6ea9fddc2db2 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\15\00\00\00\0D\00\00\00" }>, align 4 
          10: @alloc_fea0b74bc1f394cd9f32c70385fabd3f = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00\1C\00\00\00\0D\00\00\00" }>, align 4 
          11: @alloc_672a73527c968cb7e9717bc22e580e47 = private unnamed_addr constant <{ ptr, [12 x i8] }> <{ ptr @alloc_4adda3ed1088f74f07ee43568cdd9f39, [12 x i8] c";\00\00\00#\00\00\00\0D\00\00\00" }>, align 4 
          12:  
          13: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::alloc 
          14: ; Function Attrs: cold noreturn nounwind 
          15: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$5alloc17h1fa54a43d9d55781E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          16: start: 
          17: ; call std::panicking::begin_panic 
          18:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_b770eaa3f4771da96efd94be0aca7ca5) #7 
          19:  unreachable 
          20: } 
          21:  
          22: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::dealloc 
          23: ; Function Attrs: cold noreturn nounwind 
          24: define dso_local void @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$7dealloc17hb5c6154adcc1b844E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, ptr noundef readnone captures(none) %ptr, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          25: start: 
          26: ; call std::panicking::begin_panic 
          27:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_a5aa75597376cd18f9df6ea9fddc2db2) #7 
          28:  unreachable 
          29: } 
          30:  
          31: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::realloc 
          32: ; Function Attrs: cold noreturn nounwind 
          33: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$7realloc17hb1272095cf277bc4E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, ptr noundef readnone captures(none) %ptr, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1, i32 noundef %new_size) unnamed_addr #0 { 
          34: start: 
          35: ; call std::panicking::begin_panic 
          36:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_fea0b74bc1f394cd9f32c70385fabd3f) #7 
          37:  unreachable 
          38: } 
          39:  
          40: ; <global_allocator_attributes::foobar::Allocator as core::alloc::global::GlobalAlloc>::alloc_zeroed 
          41: ; Function Attrs: cold noreturn nounwind 
          42: define dso_local noalias noundef nonnull ptr @"_ZN99_$LT$global_allocator_attributes..foobar..Allocator$u20$as$u20$core..alloc..global..GlobalAlloc$GT$12alloc_zeroed17h48656aee732bdc33E"(ptr noalias noundef nonnull readonly align 1 captures(none) %self, i32 noundef range(i32 1, -2147483647) %layout.0, i32 noundef %layout.1) unnamed_addr #0 { 
          43: start: 
          44: ; call std::panicking::begin_panic 
          45:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_672a73527c968cb7e9717bc22e580e47) #7 
          46:  unreachable 
          47: } 
          48:  
          49: ; __rustc::__rust_alloc 
          50: ; Function Attrs: cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0) 
next:13'0                                                                                                   X error: no match found
          51: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc12___rust_alloc(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #1 { 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:13'1                                                             ?                                                                                                     possible intended match
          52: start: 
next:13'0     ~~~~~~~
          53:  %0 = add i32 %align, -1 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          54:  %1 = icmp sgt i32 %0, -1 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          55:  tail call void @llvm.assume(i1 %1) 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          56: ; call std::panicking::begin_panic 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          57:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_b770eaa3f4771da96efd94be0aca7ca5) #7 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          58:  unreachable 
next:13'0     ~~~~~~~~~~~~~
          59: } 
next:13'0     ~~
          60:  
next:13'0     ~
          61: ; __rustc::__rust_dealloc 
next:13'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          62: ; Function Attrs: cold noreturn nounwind allockind("free") 
next:20'0                                                               X error: match failed for invalid pattern
next:20'1                                                                 undefined variable: SIZE
next:20'2                                                                 undefined variable: SIZE
          63: define dso_local void @_RNvCs59eiacZUMsN_7___rustc14___rust_dealloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 noundef %align) unnamed_addr #2 { 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:20'3                                        ?                                                                                                                                             possible intended match
          64: start: 
next:20'0     ~~~~~~~
          65:  %0 = add i32 %align, -1 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          66:  %1 = icmp sgt i32 %0, -1 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          67:  tail call void @llvm.assume(i1 %1) 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          68: ; call std::panicking::begin_panic 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          69:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_a5aa75597376cd18f9df6ea9fddc2db2) #7 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          70:  unreachable 
next:20'0     ~~~~~~~~~~~~~
          71: } 
next:20'0     ~~
          72:  
next:20'0     ~
          73: ; __rustc::__rust_realloc 
next:20'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          74: ; Function Attrs: cold noreturn nounwind allockind("realloc,aligned") allocsize(3) 
next:27'0                                                                                       X error: match failed for invalid pattern
next:27'1                                                                                         undefined variable: SIZE
next:27'2                                                                                         undefined variable: SIZE
next:27'3                                                                                         undefined variable: SIZE
          75: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc14___rust_realloc(ptr allocptr noundef readnone captures(none) %ptr, i32 noundef %size, i32 allocalign noundef %align, i32 noundef %new_size) unnamed_addr #3 { 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          76: start: 
next:27'0     ~~~~~~~
          77:  %0 = add i32 %align, -1 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          78:  %1 = icmp sgt i32 %0, -1 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          79:  tail call void @llvm.assume(i1 %1) 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          80: ; call std::panicking::begin_panic 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          81:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_fea0b74bc1f394cd9f32c70385fabd3f) #7 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          82:  unreachable 
next:27'0     ~~~~~~~~~~~~~
          83: } 
next:27'0     ~~
          84:  
next:27'0     ~
          85: ; __rustc::__rust_alloc_zeroed 
next:27'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          86: ; Function Attrs: cold noreturn nounwind allockind("alloc,zeroed,aligned") allocsize(0) 
next:34'0                                                                                            X error: match failed for invalid pattern
next:34'1                                                                                              undefined variable: SIZE
next:34'2                                                                                              undefined variable: SIZE
          87: define dso_local noalias noundef nonnull ptr @_RNvCs59eiacZUMsN_7___rustc19___rust_alloc_zeroed(i32 noundef %size, i32 allocalign noundef %align) unnamed_addr #4 { 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
next:34'3                                                          ?                                                                                                               possible intended match
          88: start: 
next:34'0     ~~~~~~~
          89:  %0 = add i32 %align, -1 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~
          90:  %1 = icmp sgt i32 %0, -1 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          91:  tail call void @llvm.assume(i1 %1) 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          92: ; call std::panicking::begin_panic 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          93:  tail call void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance) @alloc_476509aa7047bd805d630c6546685797, i32 noundef 14, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16) @alloc_672a73527c968cb7e9717bc22e580e47) #7 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          94:  unreachable 
next:34'0     ~~~~~~~~~~~~~
          95: } 
next:34'0     ~~
          96:  
next:34'0     ~
          97: ; std::panicking::begin_panic 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          98: ; Function Attrs: cold minsize noinline noreturn nounwind optsize 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          99: declare dso_local void @_ZN3std9panicking11begin_panic17he864858cba9ca1e3E(ptr noalias noundef nonnull readonly align 1 captures(address, read_provenance), i32 noundef, ptr noalias noundef readonly align 4 captures(address, read_provenance) dereferenceable(16)) unnamed_addr #5 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         100:  
next:34'0     ~
         101: ; Function Attrs: mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         102: declare void @llvm.assume(i1 noundef) #6 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         103:  
next:34'0     ~
         104: attributes #0 = { cold noreturn nounwind "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         105: attributes #1 = { cold noreturn nounwind allockind("alloc,uninitialized,aligned") allocsize(0) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         106: attributes #2 = { cold noreturn nounwind allockind("free") "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         107: attributes #3 = { cold noreturn nounwind allockind("realloc,aligned") allocsize(3) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         108: attributes #4 = { cold noreturn nounwind allockind("alloc,zeroed,aligned") allocsize(0) "alloc-family"="__rust_alloc" "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         109: attributes #5 = { cold minsize noinline noreturn nounwind optsize "target-cpu"="generic" } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         110: attributes #6 = { mustprogress nocallback nofree nosync nounwind willreturn memory(inaccessiblemem: write) } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         111: attributes #7 = { noreturn nounwind } 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         112:  
next:34'0     ~
         113: !llvm.ident = !{!0} 
next:34'0     ~~~~~~~~~~~~~~~~~~~~
         114:  
next:34'0     ~
         115: !0 = !{!"rustc version 1.92.0-nightly (1932ce333 2025-09-22)"} 
next:34'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>>>>>>
------------------------------------------

---- [codegen] tests/codegen-llvm/global-allocator-attributes.rs stdout end ----

failures:
    [codegen] tests/codegen-llvm/global-allocator-attributes.rs

test result: FAILED. 693 passed; 1 failed; 237 ignored; 0 measured; 0 filtered out; finished in 12.99s
```

</details>

--- a/compiler/rustc_builtin_macros/src/global_allocator.rs
+++ b/compiler/rustc_builtin_macros/src/global_allocator.rs
@@ -85,7 +85,7 @@ impl AllocFnFactory<'_, '_> {
             body,
             define_opaque: None,
         }));
-        let item = self.cx.item(self.span, self.attrs(), kind);
+        let item = self.cx.item(self.span, self.attrs(method), kind);
         self.cx.stmt_item(self.ty_span, item)
     }
 
@@ -100,8 +100,18 @@ impl AllocFnFactory<'_, '_> {
         self.cx.expr_call(self.ty_span, method, args)
     }
 
-    fn attrs(&self) -> AttrVec {
-        thin_vec![self.cx.attr_word(sym::rustc_std_internal_symbol, self.span)]
+    fn attrs(&self, method: &AllocatorMethod) -> AttrVec {
+        let alloc_attr = match method.name {
+            sym::alloc => sym::rustc_allocator,
+            sym::dealloc => sym::rustc_deallocator,
+            sym::realloc => sym::rustc_reallocator,
+            sym::alloc_zeroed => sym::rustc_allocator_zeroed,
+            _ => unreachable!("Unknown allocator method!"),
+        };
+        thin_vec![
+            self.cx.attr_word(sym::rustc_std_internal_symbol, self.span),
+            self.cx.attr_word(alloc_attr, self.span)
+        ]
     }
 
     fn arg_ty(&self, input: &AllocatorMethodInput, args: &mut ThinVec<Param>) -> Box<Expr> {

--- a/tests/codegen-llvm/global-allocator-attributes.rs
+++ b/tests/codegen-llvm/global-allocator-attributes.rs
@@ -1,0 +1,41 @@
+//@ compile-flags: -C opt-level=3
+#![crate_type = "lib"]
+
+mod foobar {
+    use std::alloc::{GlobalAlloc, Layout};
+
+    struct Allocator;
+
+    unsafe impl GlobalAlloc for Allocator {
+        unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+            // CHECK-LABEL: ; __rustc::__rust_alloc
+            // CHECK-NEXT: ; Function Attrs: {{.*}}allockind("alloc,uninitialized,aligned") allocsize(0){{.*}}
+            // CHECK-NEXT: define{{.*}} noalias{{.*}} ptr @{{.*}}__rust_alloc(i[[SIZE:[0-9]+]] {{.*}}%size, i[[SIZE]] allocalign{{.*}} %align)
+            panic!()
+        }
+
+        unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
+            // CHECK-LABEL: ; __rustc::__rust_dealloc
+            // CHECK-NEXT: ; Function Attrs: {{.*}}allockind("free"){{.*}}
+            // CHECK-NEXT: define{{.*}} void @{{.*}}__rust_dealloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] {{.*}} %align)
+            panic!()
+        }
+
+        unsafe fn realloc(&self, ptr: *mut u8, layout: Layout, new_size: usize) -> *mut u8 {
+            // CHECK-LABEL: ; __rustc::__rust_realloc
+            // CHECK-NEXT: ; Function Attrs: {{.*}}allockind("realloc,aligned") allocsize(3){{.*}}
+            // CHECK-NEXT: define{{.*}} noalias{{.*}} ptr @{{.*}}__rust_realloc(ptr allocptr{{.*}} %ptr, i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align, i[[SIZE]] {{.*}} %new_size)
+            panic!()
+        }
+
+        unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
+            // CHECK-LABEL: ; __rustc::__rust_alloc_zeroed
+            // CHECK-NEXT: ; Function Attrs: {{.*}}allockind("alloc,zeroed,aligned") allocsize(0){{.*}}
+            // CHECK-NEXT: define{{.*}} noalias{{.*}} ptr @{{.*}}__rust_alloc_zeroed(i[[SIZE]] {{.*}} %size, i[[SIZE]] allocalign{{.*}} %align)
+            panic!()
+        }
+    }
+
+    #[global_allocator]
+    static GLOBAL: Allocator = Allocator;
+}

